File: frontend/src/shared/interfaces/review.ts
import { IUserResponse } from "./user";

export interface IReviewCreate {
  book_id: string;
  content: string;
  has_spoiler?: boolean;
}

export interface IReviewUpdate {
  content?: string;
  has_spoiler?: boolean;
}

export interface IReviewResponse {
  id: string;
  book_id: string;
  content: string;
  has_spoiler: boolean;
  created_at: string;
  updated_at: string;
  author: IUserResponse;
}

File: backend/app/schemas/review.py
import uuid

from pydantic import BaseModel, Field

from backend.app.schemas.user import UserResponse


class ReviewCreate(BaseModel):
    book_id: uuid.UUID
    content: str = Field(min_length=1, max_length=2000)
    has_spoiler: bool = False


class ReviewUpdate(BaseModel):
    content: str | None = Field(default=None, min_length=1, max_length=2000)
    has_spoiler: bool | None = None


class ReviewResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    book_id: uuid.UUID
    content: str
    has_spoiler: bool
    created_at: str
    updated_at: str
    author: UserResponse
